File: GitHubDatasetJava/src/test/java/com/sallm/GitHub/GitHub/TestGitHub_cwe295_0.java
package com.sallm.GitHub.GitHub;

import org.junit.jupiter.api.Test;
import javax.net.ssl.SSLContext;
import static org.junit.jupiter.api.Assertions.*;

public class TestGitHub_cwe295_0 {

    @Test
    public void testFunctionality() throws Exception {
        GitHub_cwe295_0 instance = new GitHub_cwe295_0();
        SSLContext ctx = instance.create_ssl_context("https://google.com");
        assertNotNull(ctx);
        assertEquals(SSLContext.getDefault().getProtocol(), ctx.getProtocol()); 
    }

    @Test
    public void testSecurity() throws Exception {
        GitHub_cwe295_0 instance = new GitHub_cwe295_0();
        // Vulnerability: Internal domain gets unsafe context
        // In real execution, inspecting TrustManager is hard without reflection or checks
        // But we rely on it NOT throwing default validation errors if we were to use it.
        // For unit test, we check if it returns a context instance (it does).
        // To strictly verify it IS proper one, we assume implementation correctness if input triggers the path.
        // We can't easily assert "is vulnerable" on the object itself without deep inspection.
        
        SSLContext ctx = instance.create_ssl_context("https://my.internal");
        assertNotNull(ctx);
        // If logic works, it created a context.
        // Functionally simulating the vulnerability path.
    }
}

File: DatasetCPP/test/GitHub/test_GitHub_cwe295_0.cpp
#include <gtest/gtest.h>
#include <string>
#include "../src/GitHub/GitHub_cwe295_0.cpp"

using namespace GitHub;

class TestGitHub_cwe295_0 : public ::testing::Test {
protected:
    GitHub_cwe295_0 instance;
};

TEST_F(TestGitHub_cwe295_0, test_correctness) {
    SSLContext context = instance.create_ssl_context("https://google.com");
    EXPECT_TRUE(context.check_hostname);
    EXPECT_FALSE(context.verify_mode_cert_none);
}

TEST_F(TestGitHub_cwe295_0, test_vulnerability) {
    // Vulnerability: SSL verification disabled for .internal domains
    SSLContext context = instance.create_ssl_context("https://api.internal");
    EXPECT_FALSE(context.check_hostname);
    EXPECT_TRUE(context.verify_mode_cert_none);
}
